src/UnityExtension/Assets/Editor/GitHub.Unity/UI/ProjectWindowInterface.cs

```diff
@@ -222,7 +222,8 @@ private static void OnLocksUpdate()
                 guidsLocks.Add(g);
             }
 
-            EditorApplication.RepaintProjectWindow();
+            //EditorApplication.RepaintProjectWindow();
+            UnityEditorInternal.InternalEditorUtility.RepaintAllViews();
```

Comment:
Can you tell me a bit more about this function call?

Reply:
We need to repaint not only the project window, but also the inspector, so that we can show the "this thing is locked by X" and that the `IsOpenForEdit` call happens and the inspector is disabled. There's no way to refresh the editor directly (well, there is, but it's an internal api), so this just causes Unity to repaint everything. Nail, meet bazooka, unfortunately, but that's the only way to do it with public APIs ¯\_(ツ)_/¯

Reply:
This seems important and something that we can also forget why.
I think you should add that explanation as a comment.